tensilelite/Tensile/SolutionStructs.py

```diff
@@ -2894,6 +2939,32 @@ def calcOptGRVW(lrvw: int, unrollMajorLDS: bool, datatype: DataType) -> int:
         if state["EnableMatrixInstruction"] and globalParameters["AsmCaps"][isa]['HasWMMA']:
           reject(state, "Half WMMA doesn't support single buffer GSU")
           return
+    # dot2 limitation
+    if state["UseDotInstruction"]:
+      if state["EnableMatrixInstruction"]:
+        reject(state, "dot inst is for mac kernel!")
+      if not (state["ProblemType"]["DataType"].isHalf() and state["ProblemType"]["HighPrecisionAccumulate"]):
+        reject(state, "dot2 kernel only support DataType fp16 with HPA")
+      if state["InnerUnroll"] not in [1,2,4]:
+        reject(state, "dot2 kernel requires InnerUnroll = 1,2 or 4")
+      if state["NumWaveSplitK"] not in [1,2,4,8,16,32,64]:
+        reject(state, "Unsupported NumWaveSplitK value. Need to be power of 2 and does not exceed 64.")
+      if state["DepthU"] % (state["LocalReadVectorWidth"] * state["NumWaveSplitK"]) != 0:
+        reject(state, "Non-valid DepthU for dot2 kernel, need to be multiple of (LocalReadVectorWidth * InnerUnroll * NumWaveSplitK) for atomics")
+      if state["VectorWidthA"] != 1 or state["VectorWidthB"] != 1:
+        reject(state, "dot2 kernel requires VectorWidth = 1")
+      if (state["ThreadTile0"] != 1 or state["ThreadTile1"] != 1) and state["InnerUnroll"] > 1: 
+        reject(state, "dot2 kernel does not support wider local read with ThreadTile > 1")
+      if state["ScheduleLocalWrite"] != 1:
+        reject(state, "dot2 kernel requires ScheduleLocalWrite = 1")
+      if state["LocalSplitU"] != 1:
+        reject(state, "dot2 kernel requires LocalSplitU = 1")
+      coalescedA = min(state["AssertFree0ElementMultiple"], state["MacroTile0"]) if state["ProblemType"]["TLUA"] else state["DepthU"]
+      coalescedB = min(state["AssertFree1ElementMultiple"], state["MacroTile1"]) if state["ProblemType"]["TLUB"] else state["DepthU"]
+      if coalescedA % state["GlobalReadVectorWidthA"] != 0 or coalescedB % state["GlobalReadVectorWidthB"] != 0:
+        reject(state, "dot2 kernel doesn't support edge read currently")
```

Comment:
edge is dependent with problem size. We don't need to reject it here. Just make sure that the AssertFreeXElementMultiple be the correct number.
ex:
state["AssertFree0ElementMultiple"] = max(state["AssertFree0ElementMultiple"],state["GlobalReadVectorWidthA"])